### PR TITLE
calendar field not triggering onchange after clear button pressed

### PR DESCRIPTION
**Pull request by hikashop-nicolas**
Pull Request for Issue #27054 .

### Summary of Changes
We add the trigger to the onchange event after clearing the input field when the "clear" button is pressed.

### Testing Instructions
install HikaShop Starter
go to Components->HikaShop->Orders
select a date in one of the calendar field on the top right of the listing
the page is refreshed with the calendar field selected date set
click on the "clear" button of that calendar field

### Expected result
the value is cleared and the page is refreshed with the calendar field empty

### Actual result
the value is cleared and the page is not refreshed

### CODE REVIEW

**Comment by Quy** [comment]
Edit an article.
Clear `Start Publishing`.

See in Console.
> TypeError: self.inputField.onchange is not a function calendar.js:801:21

**Comment by hikashop-nicolas** [comment]
Thank you for your feedback. I've added a check to call onchange only when defined. That will prevent the javascript error when the onchange is not defined, like on the Joomla article edition page.

File: media/system/js/fields/calendar.js [hunk 1]
```diff
@@ -798,6 +798,8 @@
 				self.inputField.setAttribute('data-alt-value', "0000-00-00 00:00:00");
 				self.inputField.setAttribute('value', '');
 				self.inputField.value = '';
+				if(self.inputField.onchange)
```
Comment (Quy) [inline media/system/js/fields/calendar.js]: Please add a space after `if` and opening/closing curly brackets. 
https://developer.joomla.org/coding-standards/javascript-j3.html#conditional-statements

**Comment by hikashop-nicolas** [comment]
It's done

File: media/system/js/fields/calendar.js [hunk 2]
```diff
@@ -798,6 +798,9 @@
 				self.inputField.setAttribute('data-alt-value', "0000-00-00 00:00:00");
 				self.inputField.setAttribute('value', '');
 				self.inputField.value = '';
+				if(self.inputField.onchange) {
```
Comment (Quy) [inline media/system/js/fields/calendar.js]:
```suggestion
				if (self.inputField.onchange) {
```

**Comment by Quy** [comment]
I have tested this item :white_check_mark: successfully on 49090260f583f9902706c3aec8e7207466d1d70e<br /><br />Also can be tested here: Components > Banners > Tracks > Search Tools<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27055">issues.joomla.org/tracker/joomla-cms/27055</a>.</sub>

**Comment by viocassel** [comment]
I have tested this item :white_check_mark: successfully on 49090260f583f9902706c3aec8e7207466d1d70e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27055">issues.joomla.org/tracker/joomla-cms/27055</a>.</sub>

**Comment by alikon** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/27055">issues.joomla.org/tracker/joomla-cms/27055</a>.</sub>

**Comment by HLeithner** [comment]
Thanks for the for fix.

**Comment by Quy** [comment]
@hikashop-nicolas Can you please check the minified file as the calendar is not displaying?

> SyntaxError: expected expression, got keyword 'if' calendar.min.js:1:15600

**Comment by Quy** [comment]
See PR #27233

**Pull request merged**